client.go

```diff
@@ -346,6 +363,34 @@ func WithOAuth() ClientOptions {
 	}
 }
 
+func WithScopedOAuthApp(clientId string, clientSecret string, scopes []string) ClientOptions {
```

Comment:
Since support for User Scoped OAuth token will be also added and this is a public API, I decided to distinguish it from the eventual implementation of `WithScopedOAuthUser` in advance. Due to `WithScopedOAuthUser` will use `oauth2.Config` and this `oauth2.Credentials`. I did the same distinction with `WithScopedOAuthAppTokenSource`.